### PR TITLE
Split GetMutableState API into two APIs

### PR DESCRIPTION
**Pull request by yux0**
* GetMutableState will remains as is, and serve the purpose of get mutable state instantly
* PollMutableState allows long poll on mutable state changes

### CODE REVIEW

File: idl/github.com/uber/cadence/history.thrift [hunk 1]
```diff
@@ -318,8 +319,25 @@ service HistoryService {
...
       6: shared.ServiceBusyError serviceBusyError,
+      7: shared.CurrentBranchChangedError currentBranchChangedError,
     )
 
+  /**
+   * Returns the information from mutable state of workflow execution.
+   * It fails with 'EntityNotExistError' if specified workflow execution in unknown to the service.
+   * It returns CurrentBranchChangedError if the workflow version branch has changed.
+   **/
+   GetMutableStateResponse GetMutableStateWithLongPoll(1: GetMutableStateRequest getRequest)
```
Comment (wxing1292) [inline idl/github.com/uber/cadence/history.thrift]: GetMutableStateWithLongPoll -> PollMutableState?

File: service/history/handler.go [hunk 2]
```diff
@@ -713,6 +713,40 @@ func (h *Handler) GetMutableState(ctx context.Context,
 	return resp, nil
 }
 
+// GetMutableStateWithLongPoll - returns the id of the next event in the execution's history
+func (h *Handler) GetMutableStateWithLongPoll(ctx context.Context,
```
Comment (wxing1292) [inline service/history/handler.go]: indentation

File: idl/github.com/uber/cadence/history.thrift (hunk 1 above)
Comment (wxing1292) [inline idl/github.com/uber/cadence/history.thrift]: input and output should be a different type

Comment (yux0) [reply idl/github.com/uber/cadence/history.thrift]: updated

File: service/history/handler.go (hunk 2 above)
Comment (yux0) [reply service/history/handler.go]: done

File: idl/github.com/uber/cadence/history.thrift (hunk 1 above)
Comment (yux0) [reply idl/github.com/uber/cadence/history.thrift]: separated the request and response

File: idl/github.com/uber/cadence/history.thrift [hunk 3]
```diff
@@ -86,6 +86,31 @@ struct GetMutableStateResponse {
...
+  35: optional i64 (js.type = "Long") PreviousStartedEventId
+  40: optional i64 (js.type = "Long") LastFirstEventId
+  50: optional shared.TaskList taskList
+  60: optional shared.TaskList stickyTaskList
+  70: optional string clientLibraryVersion
+  80: optional string clientFeatureVersion
+  90: optional string clientImpl
+  100: optional bool isWorkflowRunning
+  110: optional i32 stickyTaskListScheduleToStartTimeout
+  120: optional i32 eventStoreVersion
```
Comment (wxing1292) [inline idl/github.com/uber/cadence/history.thrift]: remove this

File: idl/github.com/uber/cadence/history.thrift [hunk 4]
```diff
@@ -86,6 +86,31 @@ struct GetMutableStateResponse {
...
+  20: optional shared.WorkflowType workflowType
+  30: optional i64 (js.type = "Long") NextEventId
+  35: optional i64 (js.type = "Long") PreviousStartedEventId
+  40: optional i64 (js.type = "Long") LastFirstEventId
+  50: optional shared.TaskList taskList
+  60: optional shared.TaskList stickyTaskList
+  70: optional string clientLibraryVersion
+  80: optional string clientFeatureVersion
+  90: optional string clientImpl
+  100: optional bool isWorkflowRunning
```
Comment (wxing1292) [inline idl/github.com/uber/cadence/history.thrift]: plz change this to workflow state
also add a workflow close status

File: idl/github.com/uber/cadence/history.thrift [hunk 5]
```diff
@@ -86,6 +86,31 @@ struct GetMutableStateResponse {
...
+  110: optional i32 stickyTaskListScheduleToStartTimeout
+  120: optional i32 eventStoreVersion
+  130: optional binary branchToken
```
Comment (wxing1292) [inline idl/github.com/uber/cadence/history.thrift]: name this current branch token

File: idl/github.com/uber/cadence/history.thrift (hunk 3 above)
Comment (yux0) [reply idl/github.com/uber/cadence/history.thrift]: deleted

File: idl/github.com/uber/cadence/history.thrift (hunk 4 above)
Comment (yux0) [reply idl/github.com/uber/cadence/history.thrift]: Updated. Please take another look

File: idl/github.com/uber/cadence/history.thrift (hunk 5 above)
Comment (yux0) [reply idl/github.com/uber/cadence/history.thrift]: Will include in the current branch PR

**Pull request merged**